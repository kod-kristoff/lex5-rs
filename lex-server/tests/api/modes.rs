use reqwest::StatusCode;
use rstest::rstest;
use serde_json::Value as JsonValue;

use crate::conftest::TestApp;

#[tokio::test]
async fn test_modes_returns_200() -> anyhow::Result<()> {
    // Arrange
    let app = TestApp::spawn_app().await?;
    let client = reqwest::Client::new();
    // Act
    let response = client
        // Use the returned application address
        .get(app.modes())
        .send()
        .await?;

    // Assert
    assert_eq!(response.status(), StatusCode::OK);
    let data: JsonValue = response.json().await?;
    insta::assert_json_snapshot!(data);
    Ok(())
}

#[rstest]
#[case("panacea")]
#[tokio::test]
async fn test_modeinfo_returns_200(#[case] mode: &str) -> anyhow::Result<()> {
    // Arrange
    let app = TestApp::spawn_app().await?;
    let client = reqwest::Client::new();
    // Act
    let response = client.get(app.modeinfo(mode)).send().await?;

    // Assert
    assert_eq!(response.status(), StatusCode::OK);
    let data: JsonValue = response.json().await?;
    insta::assert_json_snapshot!(format!("test_modeinfo_returns_200-{mode}"), data);
    Ok(())
}
