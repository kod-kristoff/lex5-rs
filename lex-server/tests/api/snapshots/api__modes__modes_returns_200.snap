---
source: lex-server/tests/api/modes.rs
expression: data
---
{
  "default": [],
  "foo": [],
  "karp": [
    "panacea"
  ],
  "large_lex": [],
  "panacea": [],
  "panacea_links": [
    "panacea"
  ]
}
